docs/developer-docs/defi/tokens/ledger/usage/icp_ledger_usage.mdx

```diff
@@ -0,0 +1,297 @@
+let ledger_id = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+// The request object of the `icrc1_name` endpoint is empty.
+    let req = ();
+    let (res,): (String,) =
+        ic_cdk::call(ledger_id, "icrc1_name", (req,))
+            .await.unwrap();
+```
+
+For all other endpoints you can use the request and response structure from the `ledger.did` Candid file. How to retrieve the Candid file is explained in [the guide on deploying an ICP ledger locally](/docs/developer-docs/defi/tokens/ledger/setup/icp_ledger_setup.mdx).
+
+### `icrc-ledger-types` Rust crate
+As explained in the [token standards](/docs/current/developer-docs/defi/tokens/token-standards), the ICP ledger supports all ICRC-1 endpoints. You will need to define the structures used for these endpoints.
+To interact with ICRC-1 and ICRC-2 endpoints, the Rust crate [icrc-ledger-types](https://crates.io/crates/icrc-ledger-types) can be used.
+This is true for the ICP ledger as well as any other canister that supports ICRC-1 or any of the ICRC-1 extension standards (i.e ICRC-2, ICRC-3,...).
+The crate can be installed with the command:
+
+```
+cargo add icrc-ledger-types
+```
+
+Or, it can be added to the `Cargo.toml` file:
+
+```
+icrc-ledger-types = "0.1.1"
+```
+
+[View the documentation for this crate](https://docs.rs/icrc-ledger-types/0.1.1/icrc_ledger_types/).
+
+
+### Receiving ICP
+
+If you want a canister to receive payment in ICP you need to make sure that the canister knows about the payment, because a transfer only involves the sender and the ledger canister.
+
+There are currently two main patterns to achieve this. Furthermore, there is a [chartered working group](https://forum.dfinity.org/t/announcing-technical-working-groups/11781) on Ledger & Tokenization which is focused on defining a standard ledger/token interface as well payment flows.
+
+#### Direct notification by sender
+
+In this pattern the sender notifies the receiver about the payment. However, the receiver needs to verify the payment by using the [`query_blocks` interface](/docs/current/references/ledger#_getting_ledger_blocks) of the ledger.
+The following diagram shows a simplified illustration of this pattern:
+
+```plantuml
+    participant Sender
+    participant "ICP Ledger"
+    participant Receiver
+
+    Sender -> "ICP Ledger": transfer()
+    "ICP Ledger" --> Sender: blockNumber
+    Sender -> Receiver: notify(blockNumber)
+    Receiver -> "ICP Ledger": query_blocks(blockNumber)
+    "ICP Ledger" --> Receiver: block
+    Receiver -> Receiver: verify payment
+```
+
+
+#### Notification by ICP ledger (currently disabled)
+
+In this pattern the ledger itself notifies the receiver. Thereby, the receiver can trust the notification immediately. However, this flow is currently disabled because the call to the receiver is not yet implemented as a one-way call.
```

Comment:
```suggestion
In this pattern, the ledger itself notifies the receiver. Thereby, the receiver can trust the notification immediately. However, this flow is currently disabled because the call to the receiver is not yet implemented as a one-way call.
```